### PR TITLE
Added unit tests for ShardingRemoveTokenGenerator#isGenerateSQLToken

### LINKED ISSUE #16093: Add unit test for ShardingRemoveTokenGenerator
Hi community,

We need to add some unit test for ShardingRemoveTokenGenerator.

- [X] ShardingRemoveTokenGenerator#isGenerateSQLToken
- [x] ShardingRemoveTokenGenerator#generateSQLTokens

### PR DESCRIPTION
**Pull request by Rohitpandey357**
Fixes #16093 (Partially) .

Changes proposed in this pull request:
- Added unit tests for ShardingRemoveTokenGenerator#isGenerateSQLToken.

### CODE REVIEW

**Comment by Rohitpandey357** [comment]
Hello all, build is failing because of checkstyle violations. How can I check for my checkstyle errors? Where can I see, what are the actual errors?

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/rewrite/token/ShardingRemoveTokenGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    
+    @Test
+    public void assertIsGenerateSQLToken() {
+        ShardingRemoveTokenGenerator shardingRemoveTokenGenerator = new ShardingRemoveTokenGenerator();
+        assertFalse(shardingRemoveTokenGenerator.isGenerateSQLToken(mock(InsertStatementContext.class)));
+        SelectStatementContext selectStatementContext = mock(SelectStatementContext.class, RETURNS_DEEP_STUBS);
+        when(selectStatementContext.getProjectionsContext().getAggregationDistinctProjections().isEmpty()).thenReturn(Boolean.TRUE);
+        assertFalse(shardingRemoveTokenGenerator.isGenerateSQLToken(selectStatementContext));
+        when(selectStatementContext.getProjectionsContext().getAggregationDistinctProjections().isEmpty()).thenReturn(Boolean.FALSE);
+        assertTrue(shardingRemoveTokenGenerator.isGenerateSQLToken(selectStatementContext));
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/rewrite/token/ShardingRemoveTokenGeneratorTest.java:41]: Please split this method into 2 methods. One method should only contains one assertion.
There are 2 calls for isGenerateSQLToken().

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/rewrite/token/ShardingRemoveTokenGeneratorTest.java (hunk 1 above)
Comment (Rohitpandey357) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/rewrite/token/ShardingRemoveTokenGeneratorTest.java:41]: Hi @terrymanu , I could do that but i just followed the project structure. In all the other test classes, there are 2 calls for isGenerateSQLToken(). If i split this in 2 methods, then we should also split in the other test classes. 

For example, refer to this -> https://github.com/apache/shardingsphere/blob/master/shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/rewrite/token/ShardingInsertValuesTokenGeneratorTest.java

Comment (terrymanu) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/rewrite/token/ShardingRemoveTokenGeneratorTest.java:41]: Yes, each assertion should assert one scenario. It is better to split all of them

Comment (Rohitpandey357) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/rewrite/token/ShardingRemoveTokenGeneratorTest.java:41]: Okay, working on it

**Comment by Rohitpandey357** [comment]
Hi @terrymanu , please have a look now.

**Comment by terrymanu** [comment]
> Hi @terrymanu , please have a look now.

Look good for me now

**Review by terrymanu — APPROVED** [review]

**Pull request merged**